File: cpp/algorithms/BogoSort.cpp
bool isSorted(int a[], int n) 
{ 
    while ( --n > 1 ) 
        if (a[n] < a[n-1]) 
            return false; 
    return true; 
} 

void shuffle(int a[], int n) 
{ 
    for (int i=0; i < n; i++) 
        swap(a[i], a[rand()%n]); 
} 
 
void bogosort(int a[], int n) 
{ 
    while ( !isSorted(a, n) ) 
        shuffle(a, n); 
} 

File: python/algorithms/bogosort.py
#https://github.com/DoctorLai/Algorithms/blob/master/Sorting/BogoSort.py
#https://helloacm.com/bogo-sort-algorithm/

#!/usr/bin/env python
"""
    http://acm.zhihua-lai.com
    BogoSort.py
    Bogo Sorting Algorithm
    30/May/2012
"""

from random import *
from time import *

seed()
x = []
for i in range(0, 10):
    x.append(randint(0, 100))

def inorder(x):
    i = 0
    j = len(x)
    while i + 1 < j:
        if x[i] > x[i + 1]:
            return False
        i += 1
    return True

def bogo(x):
    while not inorder(x):
        shuffle(x)
    return x

start = time()
print "Before: ", x
x = bogo(x)
print "After: ", x
print "%.2f seconds" % (time() - start)
